internal/testutil/api_deleter.go

```diff
@@ -0,0 +1,60 @@
+package testutil
+
+import (
+	"context"
+	"time"
+
+	"connectrpc.com/connect"
+)
+
+// WithIntervalRetry repeatedly executes the provided function `f` at the specified `interval`
+// until it returns true or the context `ctx` is done. It uses a ticker to manage the interval
+// timing. If the context is canceled or times out, the function will return without retrying further.
+//
+// Parameters:
+// - ctx: The context to control the lifecycle of the retries. If the context is done, the retries stop.
+// - interval: The duration to wait between each retry attempt.
+// - f: The function to be executed repeatedly. It should return true to stop further retries, or an error to denote success.
+func WithIntervalRetry(ctx context.Context, interval time.Duration, f func() (bool, error)) error {
```

Comment:
I think we can probably merge this function into wait for delete - this way the caller can set an appropriate interval themselves, as the time it takes to delete something can depend on the type resource itself

If you anticipate us using this retry logic more, you can pull that out into a separate private function - but I think the end user should pass the interval as well. Very clever to have them also pass a generic connect request since deletions are represented by 404s